gcp-observability/src/main/java/io/grpc/gcp/observability/GcpObservability.java

```diff
@@ -97,13 +97,8 @@ static GcpObservability grpcInit(
       InternalLoggingServerInterceptor.Factory serverInterceptorFactory)
       throws IOException {
     if (instance == null) {
-      instance =
-          new GcpObservability(sink, config, channelInterceptorFactory, serverInterceptorFactory);
-      LogHelper logHelper = new LogHelper(sink, TimeProvider.SYSTEM_TIME_PROVIDER);
-      ConfigFilterHelper logFilterHelper = ConfigFilterHelper.factory(config);
-      instance.setProducer(
-          new InternalLoggingChannelInterceptor.FactoryImpl(logHelper, logFilterHelper),
-          new InternalLoggingServerInterceptor.FactoryImpl(logHelper, logFilterHelper));
+      instance = new GcpObservability(sink, config);
+      instance.setProducer(channelInterceptorFactory, serverInterceptorFactory);
```

Comment:
Something that's now clear is that we only ever create one instance of the `InternalLoggingChannelInterceptor` and `InternalLoggingServerInterceptor` each and install those in the Global interceptors. So the factory is kind of useless. One thing we may consider (may be a separate PR) is to eliminate the Factories and replace those with those respective interceptors.

Reply:
How about a TODO comment saying we will eliminate the factory?